src/main/java/com/truelayer/java/http/OkHttpClientFactory.java

```diff
@@ -66,12 +61,22 @@ public OkHttpClient buildAuthApiClient(
                     new HttpLoggingInterceptor(logMessageConsumer, new SensitiveHeaderGuard()));
         }
 
-        clientBuilder.addInterceptor(new IdempotencyKeyInterceptor());
```

Comment:
I agree on removing the idempotency key generation from the auth client. Whilst not problematic, it was probably an oversight 😅 

That said, the auth client is currently used as base client for the `commonApiHandler`. The reason is that we need, [as per this doc](https://docs.truelayer.com/reference/submit-payments-provider-return-parameters) at least, idempotency keys generated on the provider return params submission call.

If acceptance tests in sandbox are not failing though, it might be the case that idempotency headers are not required on that endpoint... I'd take the chance to review our API's expectations and move on accordingly here